File: iteration-2/src/ProjectChecker.java
public class ProjectChecker extends Checker{
    private Course course;

    public ProjectChecker(Student student, Course course) {
        super(student);
        this.course = course;
    }

    public Course getCourse() {return course;}

    public void setCourse(Course course) {this.course = course;}

    @Override
    public Course raiseCourse() {return course;}

    @Override
    public String checkerMessage() {return course.getCourseId() + " because student completed credits < 165";}
}

File: iteration-3/ProjectChecker.py
from Checker import Checker

class ProjectChecker(Checker):
    def __init__(self, student, course):
        super().__init__(student)
        self.course = course

    def get_course(self):
        return self.course

    def set_course(self, course):
        self.course = course

    def raise_course(self):
        return self.course

    def raise_error(self):
        return "The advisor disapproved  " + self.course.get_course_id() + " since the student earned credits < 165"